src/Microsoft.ML.Core/Utilities/ResourceManagerUtils.cs

```diff
@@ -328,4 +318,25 @@ public static ResourceDownloadResults GetErrorMessage(out string errorMessage, p
         private static extern int chmod(string pathname, int mode);
 #pragma warning restore IDE1006
     }
+
+    public class WebClientResponseUri : WebClient
+    {
+        public Uri ResponseUri { get; private set; }
+
+        protected override WebResponse GetWebResponse(WebRequest request)
+        {
+            WebResponse response = null;
+
+            try
+            {
+                response = base.GetWebResponse(request);
+                ResponseUri = response.ResponseUri;
+            }
+            catch
```

Comment:
Why are you catching and eating all exceptions here?

Reply:
I think the Try/Catch is not needed. 

Apologies. This is my fault for [recommending](https://github.com/dotnet/machinelearning/pull/4943#pullrequestreview-374711517) the [idea src](http://www.nullskull.com/a/1484/make-the-webclient-class-follow-redirects-and-get-target-url.aspx) a bit too closely. 

We're not calling the original `WebRequest.ResponseUri` ([src](https://github.com/microsoft/referencesource/blob/3b1eaf5203992df69de44c783a3eda37d3d4cd10/System/net/System/Net/WebResponse.cs#L193-L217)), so no exceptions to worry about from that line. Its docs say to override in a derived class to get the redirected URI ([docs](https://docs.microsoft.com/en-us/dotnet/api/system.net.webresponse.responseuri?view=netcore-3.1#System_Net_WebResponse_ResponseUri)).

I don't see direct documentation on which exceptions `WebRequest.GetWebResponse()` can throw, could dig deeper in its [src](https://github.com/dotnet/runtime/blob/036fb0779a5bc89026d2ec64172897ab86af5828/src/libraries/System.Net.WebClient/src/System/Net/WebClient.cs#L247-L252). Only reason I would see to catch for this is if we wanted more specific logging; anything else?